main/rtp_engine.c

```diff
@@ -1513,6 +1527,23 @@ struct ast_rtp_payload_type *ast_rtp_codecs_get_payload(struct ast_rtp_codecs *c
 	return type;
 }
 
+struct ast_format *ast_rtp_codecs_get_preferred(struct ast_rtp_codecs *codecs)
+{
+	struct ast_format *format;
+	ast_rwlock_rdlock(&codecs->codecs_lock);
+	format = ao2_bump(codecs->preferred_format);
+	ast_rwlock_unlock(&codecs->codecs_lock);
+	return format;
+}
+
+int ast_rtp_codecs_set_preferred(struct ast_rtp_codecs *codecs, struct ast_format *format)
+{
+	ast_rwlock_rdlock(&codecs->codecs_lock);
```

Comment:
you want a wrlock, not a rdlock, rdlock is for reading yo, wrlock is for writing yo